src/PublishCodeCovCoverageTask/utils/webUtils.ts

```
@@ -21,197 +19,109 @@ export async function downloadFile(
-                        percent: isNaN(totalBytes) ? null : Math.round((bytesReceived / totalBytes) * 100)
+                // Setup progress tracking manually since onDownloadProgress doesn't work with streams
+                let bytesReceived = 0;
+                if (options.onProgress) {
+                    // The chunk size here is dynamic and determined by:
+                    // 1. Node.js's HTTP client internal buffering
+                    // 2. OS network buffers and TCP packet sizes
+                    // 3. Server configuration and how it streams data
+                    // The size is not fixed and varies based on network conditions
+                    (response.data as Stream).on('data', (chunk) => {
+                        bytesReceived += chunk.length;
+                        options.onProgress!({
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)